src/raft.c

```diff
@@ -183,6 +183,126 @@ static void executeRaftRedisCommandArray(RaftRedisCommandArray *array,
     }
 }
 
+typedef enum KeysStatus{
+    NoKeys,
+    AllExist,
+    SomeExist,
+    NoneExist,
+} KeysStatus;
+
+KeysStatus validateKeyExistence(RedisRaftCtx *rr, RaftRedisCommandArray *cmds) {
```

Comment:
```c
    int total_keys = 0;
    int found = 0;

    for (int i = 0; i < cmds->len; i++) {
        RaftRedisCommand *cmd = cmds->commands[i];

        /* Iterate command keys */
        int num_keys;
        int *keyindex = RedisModule_GetCommandKeys(rr->ctx, cmd->argv, cmd->argc, &num_keys);

        total_keys += num_keys;
        for (int j = 0; j < num_keys; j++) {
            found += RedisModule_KeyExists(rr->ctx, cmd->argv[keyindex[j]]);
        }
        RedisModule_Free(keyindex);
    }

    if (found != total_keys) {
        return (found == 0) ? NoneExist : SomeExist;
    }

    return (total_keys == 0) ? NoKeys : AllExist;
```
I didn't try but something like above, would that be equivalent? Counting keys found and compare at the end. You can move this logic  into place you call function, then maybe no need for enum definitions and switch case after function call. Just consider if it makes sense. (And ofcourse if it is correct).

Reply:
Iterating all keys should not be a problem as we are going to do that for almost all operations. (For most operations all keys will be found). But yeah, you can add a shortcut too.